lib/sycamore/sycamore/transforms/summarize.py

```diff
@@ -143,7 +141,238 @@ def collapse(text: str, tokens_per_chunk: int, tokenizer: Tokenizer, summarizer_
     return cur_summary
 
 
-class DocumentSummarizer(Summarizer):
+class HeirarchicalDocumentSummarizer(Summarizer):
```

Comment:
It would have been nice to not have this all together in a single PR, this probably could have been separate.